test/pylib/scylla_cluster.py

```diff
@@ -1240,6 +1294,28 @@ async def _before_test_req(self, request) -> str:
     async def _after_test(self, _request) -> str:
         assert self.cluster is not None
         assert self.current_test_case_full_name
+        self.logger.info(self.repr_tasks_history())
+        try:
+            self.tasks_history.pop(asyncio.current_task())
+            # copy current tasks
+            tasks = [key for key in self.tasks_history.keys()]
+            # wait/cancel all other tasks in ScyllaClusterManager
+
+            for task in tasks:
+                request = self.tasks_history.pop(task)
+                if not task.done():
+                    self.logger.info("wait for task:%s, request:%s", task, request.path_qs)
+                    await asyncio.wait_for(task, timeout=120)
+        except asyncio.TimeoutError:
+            self.logger.error(f"error on waiting coro {task.get_name()}")
+            sys.exit(1)
+
+        # check on tasks leakage
+        await asyncio.sleep(0.1)
+        if self.tasks_history:
+            self.logger.error("tasks leakage found %s, STOP PYTEST" ,self.tasks_history)
+            sys.exit(1)
```

Comment:
@kbr-scylla @kostja 

Bad news,
Waiting for all tasks does not guarantee that lesf tasks in asyncio.gather will not spawn new tasks while we waiting and 
 sys.exit(1) in pytest crash test.py. itself 
So, more logic is coming

I really suggest to use https://github.com/scylladb/scylladb/pull/17781  instead all of this new complicated logic

PR https://github.com/scylladb/scylladb/pull/17781  contains only 8 lines of new logic (other in renaming and doc strings)
rather than this PR has already 100 lines of new logic without 100% guarantee that ig handle all cases of wrong asyncio.gather usages in tests/test.py

cc @xtrey

Reply:
looks like we safe now